### PR TITLE
IDisposable for NetworkPeerBehavior 

### PR DESCRIPTION
**Pull request by noescape00**

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs [hunk 1]
```diff
@@ -68,14 +62,15 @@ public void Detach()
                     return;
 
                 this.DetachCore();
-                foreach (IDisposable dispo in this.disposables)
-                    dispo.Dispose();
-
```
[the remark this answers is not in the document]
Comment (dangershony) [reply src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs]: This is actually deleted code

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs [hunk 2]
```diff
@@ -68,14 +62,15 @@ public void Detach()
...
-                this.disposables.Clear();
-                this.AttachedPeer = null;
             }
         }
 
+        ///  <inheritdoc />
+        public virtual void Dispose()
+        {
+            this.AttachedPeer = null;
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs]: Whats the point? so it GCed?

Comment (noescape00) [reply src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NetworkPeerBehavior.cs]: So behavior throws when we are using it after it's disposed. Easier to catch such design bugs that way.

**Comment by Aprogiena** [comment]
Looks good, did not run tests, no sync as well

**Comment by dangershony** [comment]
Networks synced for a few minutes looks good.

**Pull request merged**